pkg/controller/replication/conversion.go

```diff
@@ -0,0 +1,307 @@
+}
+
+func convertList(rcList *v1.ReplicationControllerList) (*extensionsv1beta1.ReplicaSetList, error) {
+	rsList := &extensionsv1beta1.ReplicaSetList{Items: make([]extensionsv1beta1.ReplicaSet, len(rcList.Items))}
+	for i := range rcList.Items {
+		rc := &rcList.Items[i]
+		_, err := convertRCtoRS(rc, &rsList.Items[i])
+		if err != nil {
+			return nil, err
+		}
+	}
+	return rsList, nil
+}
+
+func convertPut(rs *extensionsv1beta1.ReplicaSet, putRC func(*v1.ReplicationController) (*v1.ReplicationController, error)) (*extensionsv1beta1.ReplicaSet, error) {
+	rc, err := convertRStoRC(rs)
+	if err != nil {
+		return nil, err
+	}
+	newRC, err := putRC(rc)
+	if err != nil {
+		return nil, err
+	}
+	return convertRCtoRS(newRC, nil)
+}
+
+func convertRCtoRS(rc *v1.ReplicationController, out *extensionsv1beta1.ReplicaSet) (*extensionsv1beta1.ReplicaSet, error) {
+	var rsInternal extensions.ReplicaSet
+	if err := apiv1.Convert_v1_ReplicationController_to_extensions_ReplicaSet(rc, &rsInternal, nil); err != nil {
+		return nil, fmt.Errorf("can't convert ReplicationController %v/%v to ReplicaSet: %v", rc.Namespace, rc.Name, err)
+	}
+	if out == nil {
+		out = new(extensionsv1beta1.ReplicaSet)
+	}
+	if err := extensionsinternalv1beta1.Convert_extensions_ReplicaSet_To_v1beta1_ReplicaSet(&rsInternal, out, nil); err != nil {
+		return nil, fmt.Errorf("can't convert ReplicaSet (converted from ReplicationController %v/%v) from internal to extensions/v1beta1: %v", rc.Namespace, rc.Name, err)
+	}
+	return out, nil
+}
+
+func convertRStoRC(rs *extensionsv1beta1.ReplicaSet) (*v1.ReplicationController, error) {
+	var rsInternal extensions.ReplicaSet
+	if err := extensionsinternalv1beta1.Convert_v1beta1_ReplicaSet_To_extensions_ReplicaSet(rs, &rsInternal, nil); err != nil {
+		return nil, fmt.Errorf("can't convert ReplicaSet (converting to ReplicationController %v/%v) from extensions/v1beta1 to internal: %v", rs.Namespace, rs.Name, err)
+	}
+	var rc v1.ReplicationController
+	if err := apiv1.Convert_extensions_ReplicaSet_to_v1_ReplicationController(&rsInternal, &rc, nil); err != nil {
+		return nil, fmt.Errorf("can't convert ReplicaSet to ReplicationController %v/%v: %v", rs.Namespace, rs.Name, err)
+	}
+	return &rc, nil
+}
+
+type podControlAdapter struct {
+	controller.PodControlInterface
+}
+
+func (pc podControlAdapter) CreatePods(namespace string, template *v1.PodTemplateSpec, object runtime.Object) error {
+	rc, err := convertRStoRC(object.(*extensionsv1beta1.ReplicaSet))
+	if err != nil {
+		return err
+	}
+	return pc.PodControlInterface.CreatePods(namespace, template, rc)
+}
+
+func (pc podControlAdapter) CreatePodsOnNode(nodeName, namespace string, template *v1.PodTemplateSpec, object runtime.Object, controllerRef *metav1.OwnerReference) error {
+	rc, err := convertRStoRC(object.(*extensionsv1beta1.ReplicaSet))
+	if err != nil {
+		return err
+	}
+	return pc.PodControlInterface.CreatePodsOnNode(nodeName, namespace, template, rc, controllerRef)
+}
```

Comment:
`CreatePods` and `CreatePodsNoNode` aren't used by rs or rc controller

Reply:
You need to implement them to conform to the interface.